supabase/migrations/20240505115913_releases.sql

```diff
@@ -6,7 +6,12 @@ CREATE TABLE public.releases (
      created_by UUID REFERENCES public.users ON DELETE SET NULL,
      created_at TIMESTAMP WITH TIME ZONE NOT NULL,
      updated_at TIMESTAMP WITH TIME ZONE NOT NULL,
-     CONSTRAINT unique_release_title_per_project UNIQUE (project_id, release_title)
+     github_release_id BIGINT NOT NULL,
+     github_owner_slug TEXT NOT NULL,
+     github_repo_slug TEXT NOT NULL,
+     github_release_data JSON NOT NULL,
+     CONSTRAINT unique_release_title_per_project UNIQUE (project_id, release_title),
+     CONSTRAINT unique_github_release_per_project UNIQUE (project_id, github_owner_slug, github_repo_slug, github_release_id)
```

Comment:
Constraint to make sure that the same release is not used twice in the same project.

---

You really feel the need for refactoring here to have separate `github_repo` and `github_releases` tables. 

This will also make it easier to check if a single GitHub repository is not used in more than one project. In current implementation more projects could connect to the same github repository.

Reply:
Also, circling back to our discussion on Friday:
1. If a GitHub Release is created before release here, and release title here is different from GitHub Release title - what do we do? I guess we missed that scenario on Friday, but the solution is obvious - update it with the user input, right?..
2. Actually, since we are here, in general, what was the main reason for keeping release titles in sync (between release manager and GitHub)? Can't we allow them to be different? Maybe even.. we want them to be decoupled, for the following reasons. One, if I only use, only know release manager, I would be unpleasantly surprised if today I have one release title, and tomorrow another one, even though I didn't change it (but smb changed it in GitHub). Two, if I'm already using a dedicated release manager, I don't want to bother with GitHub at all, and I'm not gonna manipulate releases there.

Continuing 2, let's think about simplifying the program even more.
Because as a user, again, if I use Release Manager, what are the scenarios:
- I create releases only via Release Manager. I read releases only via Release Manager. I update releases only via Release Manager (for example, if I forgot to say in release notes that ticket XXX is also included into it). So, **I never update or delete releases via GitHub. In fact, I don't even need it to be created in GitHub. I don't care about GitHub at all.**
- I create releases in GitHub, because I have a CI/CD workflow that does that automatically after I push a new tag. That release contains a nice changelog, compared to the previous release (that's actually what I had on my project 😄). I never modify it. I then go to Release Manager and trigger release creation there. Aha. Rather than a potential checkbox "Also create a GitHub Release" **I'd rather need a checkbox or even a separate button "GitHub Release already created" which will require only git tag from me, and release title and release body will be copied from GitHub**. Now that the release is in Release Manager, similar to the previous workflow, **I never update or delete it via GitHub**. And if I want to add a couple sentences to release in Release Manager, I can do it, and I will definitely do it there, **and I don't even want it to affect GitHub Release. GitHub Release will stay sort of formal and dry there, forever.**

So, I'd say we might not need the hooks at all.

And the same in the opposite direction - if I update release in Release Manager, I don't want it to affect release in GitHub.

And when creating a release, I either want to do it fully via Release Manager (and I don't care about GitHub), or I want to create it from an already created GitHub Release (because I have a process that generates nice releases in GitHub automatically, on git tag push).